src/main/java/com/thebois/models/beings/AbstractBeing.java

```diff
@@ -13,47 +22,57 @@
 
     // The max speed of the AbstractBeing
     private static final float MAX_WALKING_DISTANCE = 0.1f;
-    // The position the AbstractBeing wants to reach.
-    private Position destination;
-    // The current position held by the AbstractBeing.
-    private Position currentPosition;
```

Comment:
Question

If the path towards the Beings destination gets interrupted by something, how does the Being remember what destination it was supposed to go to if this variable is removed?

Reply:
It knows its previous path, so when it recalculates its new path, it uses the destination of the previous path. See the `onObstaclePlaced` override method on lin 52!